packages/expo-modules-core/ios/Swift/Functions/SyncFunctionComponent.swift

```diff
@@ -82,12 +82,12 @@ public final class SyncFunctionComponent<Args, FirstArgType, ReturnType>: AnySyn
   // MARK: - JavaScriptObjectBuilder
 
   func build(inRuntime runtime: JavaScriptRuntime) -> JavaScriptObject {
-    return runtime.createSyncFunction(name, argsCount: argumentsCount) { [weak self, name] this, args in
-      guard let self = self else {
-        throw NativeFunctionUnavailableException(name)
+    return runtime.createSyncFunction(name, argsCount: argumentsCount) { [weak runtime, self] this, args in
```

Comment:
I thought it's a bit risky and may lead to memory leaks since it holds a strong reference to `self` now. These objects are "detached", we don't hold them on the native side like we do with module definitions.
I tested that against memory leaks, looks like when the GC is called on JS VM, the function definition is deallocated from the native space as expected.